sources/packages/forms/src/form-definitions/sfaa2024-25.json

```diff
@@ -5367,13 +5366,13 @@
           "properties": {},
           "lockKey": true,
           "calculateServer": true,
-          "calculateValue": "/**\r\n * Validates if the date provided is before today.\r\n *\r\n * The same validation exists on calculating selectedStudyEndDateBeforeToday,\r\n * other student application forms, program information request form\r\n * and also on the server side.\r\n *\r\n * When there is an update to this logic make sure to update in all the above.\r\n *\r\n */\r\n\r\nfunction isGivenDateBeforeToday(givenDate) {\r\n  return moment\r\n    .utc(givenDate)\r\n    .isBefore(moment.utc(new Date()).format(\"YYYY-MM-DD\"), \"day\");\r\n}\r\n\r\n// If program/study period not listed is selected, validate the study end date to be after given days before now.\r\nvalue = !!data.studyendDate && isGivenDateBeforeToday(data.studyendDate);\r\n"
+          "calculateValue": "/**\r\n * Validates if the date provided is before six weeks from today.\r\n *\r\n * The same validation exists on calculating selectedStudyEndDateBeforeSixWeeksFromToday,\r\n * other student application forms, program information request form\r\n * and also on the server side.\r\n *\r\n * When there is an update to this logic make sure to update in all the above.\r\n *\r\n */\r\n\r\nfunction isGivenDateBeforeSixWeeksFromToday(givenDate) {\r\n  const sixWeeksFromToday = new Date(Date.now() + 42*24*60*60*1000);\r\n  return moment\r\n    .utc(givenDate)\r\n    .isBefore(moment.utc(sixWeeksFromToday).format(\"YYYY-MM-DD\"), \"day\");\r\n}\r\n\r\n// If program/study period not listed is selected, validate the study end date to be after given days before now.\r\nvalue = !!data.studyendDate && isGivenDateBeforeSixWeeksFromToday(data.studyendDate);"
```

Comment:
Can the code be simplified as below?
```js
value = !!data.studyendDate && moment(data.studyendDate).isBefore(moment().subtract(6, 'week'));
````
Also, `studyendDate` is timezone agnostic. Is there a reason UTC was in the logic?

Reply:
Please consider the same for the other calculation.

Reply:
Not sure why UTC was used in the logic. I was following the original code. It makes sense to remove it as `studyendDate` is time-zone agnostic.
The code is now updated as below:
```
value =
  !!data.selectedOfferingEndDate && moment(data.studyendDate).isBefore(moment().add(6, 'week'));
```

Reply:
Just double-checking the expectation for the logic: _the study end + 6 weeks cannot be beyond today_. Is it right?

Assuming today is Sep 13, by which "study end date" the students will start to see the banner?
Is it for any study end date after August 2?

Reply:
Just to keep a record here as well.
Assume today is Sep 13th. If the "study end date" is Sep 30th, then the student will see the banner because today is within 6 weeks of the study end date. If the "study end date" is October 30th, then the student won't see the banner because today is still before 6 weeks of the study end date. The banner should be displayed if the student is applying for an offering that will end in less than six weeks.